tensorboard/webapp/widgets/card_fob/card_fob_controller_component.ts

```diff
@@ -123,10 +123,39 @@ export class CardFobControllerComponent {
     return this.axisDirection === AxisDirection.VERTICAL;
   }
 
+  private moveCurrentFob(
+    newStep: number,
+    timeSelection: TimeSelection
+  ): TimeSelection {
+    if (!this.timeSelection.end) {
+      timeSelection.start.step = newStep;
+      return timeSelection;
+    }
+
+    if (this.currentDraggingFob === Fob.END) {
+      if (newStep < this.timeSelection.start.step) {
+        timeSelection.end!.step = this.timeSelection.start.step;
+        this.currentDraggingFob = Fob.START;
+        return this.moveCurrentFob(newStep, timeSelection);
```

Comment:
Why not just set the timeSelection.start.step = newStep right here?

Reply:
I mean sure... but recursion is cool? 
![image](https://user-images.githubusercontent.com/78179109/188796254-5fb8e2cb-8c6a-4a90-a09c-8d2850033103.png)

Reply:
Recursion :skull: